packages/utils/selectors/k8s.ts

```diff
@@ -2,11 +2,14 @@ import { K8sResourceCommon } from "@openshift-console/dynamic-plugin-sdk";
 import * as _ from 'lodash';
 import { GetAPIVersionForModel } from "../../types";
 
-type GetStringProperty<T = K8sResourceCommon> = (resource: T) => string;
+type GetStringProperty<T = K8sResourceCommon & { provider?: string }> = (resource: T) => string;
 
 export const getName: GetStringProperty = (resource) =>
   resource?.metadata?.name;
 
+export const getProvider: GetStringProperty = (resource) =>
+  resource?.provider;
+
```

Comment:
check the namespacestore CR...
```suggestion
export const getProvider: GetStringProperty = (resource) =>
  resource?.spec?.type;

```

Reply:
I am not sure I understood... `resource` is a CR read it how I did in above suggestion else if it is constant object use it directly no need to create a function in utils...

Reply:
while displaying if we directly display it will be aws-s3 not AWS S3. In order to get the static keys, this function is used.